src/main/java/com/atlassian/jira/cloud/jenkins/config/JiraCloudSiteConfig2.java

```diff
@@ -167,7 +178,19 @@ public FormValidation doTestConnection(
                 return FormValidation.error("Failed to retrieve secret");
             }
 
-            // TODO (ARC-1135): call a "ping" endpoint of the Jenkins app to test the connection
+            try {
+                boolean pingSuccess = pingApi.sendPing(webhookUrl, maybeSecret.get());
+                if (!pingSuccess) {
+                    return FormValidation.error(
+                            "Connection could not be established. Is the secret correct?");
+                }
+            } catch (BadRequestException e) {
+                return FormValidation.error(
+                        String.format("Error message from Jira: %s", e.getMessage()));
+            } catch (Exception e) {
+                logger.error("Unexpected error when testing connection!", e);
+                return FormValidation.error("Unexpected error when testing connection!");
+            }
```

Comment:
This code above is sending a "ping" request to the app when the user clicks on "test settings"